src/hooks/useKeyPress.tsx

```diff
@@ -3,15 +3,19 @@
  */
 import { useEffect, useState } from 'react';
 
-function useKeyPress(targetKey: string, cb?: () => void): boolean {
+function useKeyPress(
+  targetKey: string,
+  cb?: (arg: KeyboardEvent) => void
+): boolean {
   const [keyPressed, setKeyPressed] = useState(false);
 
   useEffect(() => {
-    const downHandler = ({ key }: KeyboardEvent) => {
+    const downHandler = (event: KeyboardEvent) => {
+      const { key } = event;
       if (key === targetKey) {
         setKeyPressed(true);
         if (cb) {
-          cb();
+          cb(event);
```

Comment:
I don't think you should pass the event here.

An alternative to your solution is something:

```jsx
useKeyPress({ targetKey: 'ctrl+k', callback: expandContainer });
useKeyPress({ targetKey: 'meta+k', callback: expandContainer });
```

And in the definition:

```jsx
const decorateKey = (key: string, e: KeyboardEvent) => {
  if (e.metaKey) {
    return `meta+${key}`
  }

  if (e.ctrlKey) {
    return `ctrl+${key}`
  }

  return key;
}

interface UseKeyPressProps {
  targetKey: string;
  callback?: () => void;
  preventDefault?: boolean;
}

const useKeyPress = (props: UseKeyPressProps) => {
  const [keyPressed, setKeyPressed] = useState(false);

  useEffect(() => {
    const downHandler = ({ key, ...e }: KeyboardEvent) => {
      if (decorateKey(key) === props.targetKey) {
        if (props.preventDefault) {
          e.preventDefault();
        }

        setKeyPressed(true);

        if (typeof props.cb === 'function') {
          props.cb();
        }
      }     
    });

    const upHandler = ({ key, ...e }: KeyboardEvent) => {
      if (decorateKey(key) === props.targetKey) {
        setKeyPressed(false);
      }
    };

    window.addEventListener('keydown', downHandler);
    window.addEventListener('keyup', upHandler);

    return () => {
      window.removeEventListener('keydown', downHandler);
      window.removeEventListener('keyup', upHandler);
    };
  }, [cb, props.targetKey]);

  return keyPressed;
}
```

Reply:
The example above I made feels cleaner, decouples logic, and doesn't add multiple responsibilities to the Hook.

Reply:
Looks clean, will update the changes